### PR TITLE
[sonic-sdk] add sonic sdk and sonic sdk buildenv

### PR DESCRIPTION
**Pull request by stepanblyschak**
Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>

**- Why I did it**
To allow developers to use these images as a runtime base and development environment.

**- How I did it**
Add new targets for sonic-sdk dockers. Create Dockerfile.j2 templates for them

**- How to verify it**
Build target/sonic-sdk-buildenv.gz.
Load this image into docker.
Run this image and build sonic-swss in it.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: rules/docker-sonic-sdk-buildenv.mk [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+DOCKER_SONIC_SDK_BUILDENV_STEM = sonic-sdk-buildenv
+DOCKER_SONIC_SDK_BUILDENV = $(DOCKER_SONIC_SDK_BUILDENV_STEM).gz
+DOCKER_SONIC_SDK_BUILDENV_DBG = $(DOCKER_SONIC_SDK_BUILDENV_STEM)-$(DBG_IMAGE_MARK).gz
+
+$(DOCKER_SONIC_SDK_BUILDENV)_PATH = $(DOCKERS_PATH)/docker-sonic-sdk-buildenv
+
+$(DOCKER_SONIC_SDK_BUILDENV)_DEPENDS += $(LIBSAIVS) \
+                                        $(LIBSAIVS_DEV) \
+                                        $(LIBSAIVS_DBG) \
```
Comment (nazariig) [inline rules/docker-sonic-sdk-buildenv.mk:11]: @stepanblyschak why do we need this?

File: rules/docker-config-engine-buster.mk [hunk 2]
```diff
@@ -11,7 +11,9 @@ $(DOCKER_CONFIG_ENGINE_BUSTER)_LOAD_DOCKERS += $(DOCKER_BASE_BUSTER)
 $(DOCKER_CONFIG_ENGINE_BUSTER)_FILES += $(SWSS_VARS_TEMPLATE)
 $(DOCKER_CONFIG_ENGINE_BUSTER)_FILES += $($(SONIC_CTRMGRD)_CONTAINER_SCRIPT)
 
-$(DOCKER_CONFIG_ENGINE_BUSTER)_DBG_DEPENDS = $($(DOCKER_BASE_BUSTER)_DBG_DEPENDS)
+$(DOCKER_CONFIG_ENGINE_BUSTER)_DBG_DEPENDS = $($(DOCKER_BASE_BUSTER)_DBG_DEPENDS) \
+                                             $(LIBSWSSCOMMON_DBG) \
+                                             $(LIBHIREDIS_DBG)
```
Comment (nazariig) [inline rules/docker-config-engine-buster.mk:16]: @stepanblyschak why do we need this?

Comment (stepanblyschak) [reply rules/docker-config-engine-buster.mk:16]: hiredis library debug

File: rules/docker-sonic-sdk-buildenv.mk (hunk 1 above)
Comment (stepanblyschak) [reply rules/docker-sonic-sdk-buildenv.mk]: libsaivs-dbg moved to _DBG_DEPENDS

**Review by nazariig — APPROVED** [review]

**Pull request closed without merging**